src/plugins/contextMenu/menu.js

```diff
@@ -308,36 +349,42 @@ class Menu {
    * @param {Event} [event]
    */
   executeCommand(event) {
-    if (!this.isOpened() || !this.hotMenu.getSelectedLast()) {
+    if (!this.isOpened() || !this.hasSelectedItem()) {
       return;
     }
-    const selectedItem = this.hotMenu.getSourceDataAtRow(this.hotMenu.getSelectedLast()[0]);
+    const selectedItem = this.getSelectedItem();
 
     this.runLocalHooks('select', selectedItem, event);
 
-    if (selectedItem.isCommand === false || selectedItem.name === SEPARATOR) {
+    if (this.isCommandPassive(selectedItem)) {
       return;
     }
+
     const selRanges = this.hot.getSelectedRange();
     const normalizedSelection = selRanges ? normalizeSelection(selRanges) : [];
-    let autoClose = true;
-
-    // Don't close context menu if item is disabled or it has submenu
-    if (selectedItem.disabled === true ||
-        (typeof selectedItem.disabled === 'function' && selectedItem.disabled.call(this.hot) === true) ||
-        selectedItem.submenu) {
-      autoClose = false;
-    }
 
     this.runLocalHooks('executeCommand', selectedItem.key, normalizedSelection, event);
 
     if (this.isSubMenu()) {
       this.parentMenu.runLocalHooks('executeCommand', selectedItem.key, normalizedSelection, event);
     }
+  }
 
-    if (autoClose) {
-      this.close(true);
-    }
+  /**
+   * Checks if the passed command is passive or not. The command is passive when it's marked as
+   * disabled, the descriptor object contains `isCommand` property set to `false`, command
+   * is a separator, or the item is recognized as submenu. For passive items the menu is not
+   * closed automatically after the user trigger the command through the UI.
+   *
+   * @param {Object} commandDescriptor Selected menu item from the menu data source.
+   * @returns {Boolean}
+   */
+  isCommandPassive(commandDescriptor) {
+    const { isCommand, name: commandName, disabled, submenu } = commandDescriptor;
+
+    const isDisable = disabled === true || (typeof disabled === 'function' && disabled.call(this.hot) === true);
```

Comment:
Shouldn't this be called `isDisabled` instead of `isDisable`?

Reply:
Good catch!